### PR TITLE
Update OptionCard component

### PR DESCRIPTION
**Pull request by eldiegod**
fixes: #1676

figma: https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=4118%3A144975

### CODE REVIEW

**Review by rafalpawlow — CHANGES_REQUESTED** [review]
In figma, radio component doesn't inherit error state
Zrzut ekranu 2022-01-13 o 18 49 39

Another thing is, in figma we have two types of option card `Radio` and `Checkbox`. Second one is missing in your PR.

File: src/components/_inputs/OptionCard/OptionCard.stories.tsx [hunk 1]
```diff
@@ -21,9 +23,10 @@ const Template: Story<OptionCardProps> = (args) => {
     setSelected(element.value)
   }
   return (
-    <div style={{ display: 'grid', gridTemplateColumns: 'auto auto', gap: '16px' }}>
-      <OptionCard {...args} value="1" selectedValue={selected} onClick={handleClick} />
+    <div style={{ display: 'grid', gridTemplateColumns: 'auto auto auto', gap: '16px' }}>
```
Comment (rafalpawlow) [inline src/components/_inputs/OptionCard/OptionCard.stories.tsx]: In figma we have 2 gaps depending on the breakpoint `16px` and `24px`. Also I think we should move `Option card group` as a separate component.

**Comment by rafalpawlow** [comment]
Figma says: `In the Option card component, the display limit for both the main label is set to 1 line of text. Upon exceeding this limit, the text should be trimmed and ended with an ellipsis.`.

But current implementation allows 2 lines of text
Zrzut ekranu 2022-01-13 o 19 00 47

**Review by rafalpawlow — CHANGES_REQUESTED** [review]

**Comment by rafalpawlow** [comment]
I'm not sure about spacing between text and hovered input. @toiletgranny is that fine?
Zrzut ekranu 2022-01-13 o 19 09 48

**Comment by rafalpawlow** [comment]
Border colors are wrong
we should use the following:

**white**
default: `colorBorderAlpha`
hover: `colorBorderStrongAlpha`
pressed: `colorBorderMutedAlpha`

**error**
hover: `colorCoreRed300`
default: `colorCoreRed500`
pressed: `colorCoreRed700`

currently we use `colorCoreRed400` for all variants

**blue**
hover: `colorCoreBlue300`
pressed: `colorCoreBlue700`

**Comment by toiletgranny** [comment]
I agree with everything @rafalpawlow pointed out here, and to answer/comment on some of the points he brought up:
1. Yes, the spacing between the label and the Radio button/Checkbox should be `16px`, awesome catch! This can be inspected in Figma, though it might be tricky. 
CleanShot 2022-01-14 at 11 02 41

2. It's great to see text & icon colors using our new design tokens, but @rafalpawlow is right that we forgot about borders. Especially in "the error state". In case that's helpful, @eldiegod I put up this quick prototype for you as a quick reference on how border colors are changing upon hovering and pressing the Option card component.


https://www.figma.com/proto/Pf31tuYpozYmpq163U2ho8/Web-Components?page-id=4645%3A137109&node-id=4645%3A137110&viewport=396%2C48%2C1&scaling=min-zoom&starting-point-node-id=4645%3A137110

**Comment by eldiegod** [comment]
@rafalpawlow @toiletgranny I think I have addressed all the comments except for the optionCardGroup components as I'm not sure it should be done here or it's a matter for another ticket.

Saying it because the initial complexity scope for this ticket was a 2 but it has gotten a bit intricate so not sure if we just missed that or we are doing more than planned

**Comment by rafalpawlow** [comment]
@eldiegod component still allows for 2 lines of title

**Comment by rafalpawlow** [comment]
Something is wrong with input when there is a lot of text ant card is in `active` state
Zrzut ekranu 2022-01-17 o 15 53 11

**Comment by eldiegod** [comment]
> Something is wrong with input when there is a lot of text ant card is in `active` state

Can you tell me exactly what you mean with wrong? @rafalpawlow

**Comment by rafalpawlow** [comment]
@eldiegod Look at the screenshot, radio button is not round, it is egg-shaped

**Comment by eldiegod** [comment]
> @eldiegod Look at the screenshot, radio button is not round, it is egg-shaped

wanted to make sure, I think I fixed it now @rafalpawlow

**Review by rafalpawlow — APPROVED** [review]
Looks good! However I found one more thing which we probably should tackle in another issue. Pressed radio component looks a little bit different in figma:

Zrzut ekranu 2022-01-18 o 13 04 19
Zrzut ekranu 2022-01-18 o 13 04 57

**Comment by eldiegod** [comment]
> Looks good! However I found one more thing which we probably should tackle in another issue. Pressed radio component looks a little bit different in figma:
> 
> Zrzut ekranu 2022-01-18 o 13 04 19 Zrzut ekranu 2022-01-18 o 13 04 57

@rafalpawlow Awesome! Yep that's right, that would be more of an existing issue with the `RadioInput` component itself

**Comment by kdembler** [comment]
@eldiegod Could you please create an issue for this difference? We may have missed that

**Pull request merged**